### PR TITLE
SB-10787: Telemetry Enhancement

### PR DESCRIPTION
**Pull request by krgauraw**

### CODE REVIEW

File: searchIndex-platform/module/search-api/search-manager/app/managers/BasePlaySearchManager.java [hunk 1]
```diff
@@ -266,4 +270,37 @@ protected String getMessage(Response res) {
 		}
 		
 	}
+
+	/**
+	 * @param context
+	 * @param filters
+	 */
+	private void populateObjectField(Map<String, Object> context, Map<String, Object> filters) {
```
Comment (PradyumnaNagendra) [inline searchIndex-platform/module/search-api/search-manager/app/managers/BasePlaySearchManager.java]: can you please rephrase the method name, as this is specific to Dialcodes

File: platform-jobs/samza/audit-event-generator/src/main/resources/application.conf [hunk 2]
```diff
@@ -11,3 +11,5 @@ LearningActorSystem{
   		throughput = 1
 	}
 }
+
+channel.default = "in.ekstep"
```
Comment (PradyumnaNagendra) [inline platform-jobs/samza/audit-event-generator/src/main/resources/application.conf]: We do not need this in application.conf

File: platform-core/platform-telemetry/src/main/java/org/ekstep/telemetry/TelemetryGenerator.java [hunk 3]
```diff
@@ -184,6 +188,8 @@ public static String audit(Map<String, String> context, List<String> props, Stri
 			edata.put("state", state);
 		if (StringUtils.isNotBlank(prevState))
 			edata.put("prevstate", prevState);
+		if(StringUtils.isNotBlank(context.get("duration")))
+			edata.put("duration",Long.valueOf(context.get("duration")));
```
Comment (PradyumnaNagendra) [inline platform-core/platform-telemetry/src/main/java/org/ekstep/telemetry/TelemetryGenerator.java]: it is better to cast duration as Number and get longValue.

File: searchIndex-platform/module/search-api/search-manager/app/managers/BasePlaySearchManager.java (hunk 1 above)
Comment (krgauraw) [reply searchIndex-platform/module/search-api/search-manager/app/managers/BasePlaySearchManager.java]: changed to populateTarget
DialObject

File: platform-jobs/samza/audit-event-generator/src/main/resources/application.conf (hunk 2 above)
Comment (krgauraw) [reply platform-jobs/samza/audit-event-generator/src/main/resources/application.conf]: moved to test application.conf

File: platform-core/platform-telemetry/src/main/java/org/ekstep/telemetry/TelemetryGenerator.java (hunk 3 above)
Comment (krgauraw) [reply platform-core/platform-telemetry/src/main/java/org/ekstep/telemetry/TelemetryGenerator.java]: as discussed, no change is required here.

**Pull request merged**